tests/foreman/api/test_contentcredentials.py

```diff
@@ -227,3 +229,46 @@ def test_positive_delete(module_org):
     gpg_key.delete()
     with pytest.raises(HTTPError):
         gpg_key.read()
+
+
+@pytest.mark.tier1
+def test_positive_block_delete_key_in_use(module_org):
+    """Create a GPG key with valid content. Create a new product and
+        associated repository, assigning the GPG key to both. Attempt to delete the
+        GPG key in use.
+
+    :id: b79fd3ff-8cdb-4cdd-94e5-76de742ec967
+
+    :expectedresults: Blocked deletion of gpg key in use, it remains
+        unmodified, is still associated with product and repo.
+
+    :BZ: 2052904
+
+    :CaseImportance: Critical
+    """
+    gpg_key = entities.GPGKey(organization=module_org, content=key_content).create()
+    gpg_copy = copy(gpg_key)
+    # Create new product with gpg, and a single associated repository
+    product = entities.Product(gpg_key=gpg_key, organization=module_org).create()
+    repo = entities.Repository(product=product).create()
+    product.sync()
+    product = product.update()
+
+    # Assert the same gpg key is associated with new product and repo
+    assert product.gpg_key.id == gpg_key.id
+    assert repo.gpg_key.id == gpg_key.id
+    assert product.gpg_key.id == repo.gpg_key.id
+
+    # Attempt to delete gpg in use
+    with pytest.raises(HTTPError) as e:
+        gpg_key.delete()
+    assert len(str(e.value)) > 0
```

Comment:
Do we get any sort of useful error message here? I would prefer to assert something like "Can't delete gpg key because xyz" over asserting that there was at least one error returned. It will be easier to debug failures in the future if there is ever a regression.

Reply:
HTTPError message is `500 Server Error: Internal Server Error for url: https://ip-10-0-167-114.rhos-01.prod.psi.rdu2.redhat.com/katello/api/v2/content_credentials/204`

However nailgun shows 
`nailgun.client - WARNING - Received HTTP 500 response: {"displayMessage":"Cannot delete record because of dependent root_repositories","errors":["Cannot delete record because of dependent root_repositories"]}` -do you know how I could access these nailgun warnings?

Reply:
if you set line 264 to 
```
result = gpg_key.delete()
```
You might be able to access that error in something like `result.stderr`. I would suggest running in debug mode and looking through the result object for that error. Let me know if you need help setting that up. I can show you how to do it in VS Code if you've never done it before.